storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+
+        @Override
+        public void close() {
+            mockedWorkerUserWSE = null;
+        }
+    }
+    private static Map mockedWorkerUserWSE = null;
+    public static void setWorkerUserWSE(Map conf, String workerId, String user) throws IOException {
+        if (mockedWorkerUserWSE != null) {
+            return;
+        }
+        LOG.info("SET worker-user {} {}", workerId, user);
+        File file = new File(workerUserFile(conf, workerId));
+        file.getParentFile().mkdirs();
+
+        try (FileWriter fw = new FileWriter(file);
+             BufferedWriter writer = new BufferedWriter(fw);) {
+            writer.write(user);
+        }
+    }
+
+    public static void removeWorkerUserWSE(Map conf, String workerId) {
+        LOG.info("REMOVE worker-user {}", workerId);
+        new File(workerUserFile(conf, workerId)).delete();
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedWorkerArtifactsRoot mocked = new SetMockedWorkerArtifactsRoot(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedWorkerArtifactsRoot. root)]
+    //     run test ...)
+    public static class SetMockedWorkerArtifactsRoot implements Closeable {
+        public SetMockedWorkerArtifactsRoot(String root) {
+            mockedWorkerArtifactsRoot = root;
+        }
+
+        @Override
+        public void close() {
+            mockedWorkerArtifactsRoot = null;
+        }
+    }
+    private static String mockedWorkerArtifactsRoot = null;
+    public static String workerArtifactsRoot(Map conf) {
+        if (mockedWorkerArtifactsRoot != null) {
+            return mockedWorkerArtifactsRoot;
+        }
+        String artifactsDir = (String)conf.get(Config.STORM_WORKERS_ARTIFACTS_DIR);
+        if (artifactsDir == null) {
+            return (LOG_DIR + FILE_SEPARATOR + "workers-artifacts");
+        } else {
+            if (new File(artifactsDir).isAbsolute()) {
+                return artifactsDir;
+            } else {
+                return (LOG_DIR + FILE_SEPARATOR + artifactsDir);
+            }
+        }
+    }
+
+    public static String workerArtifactsRoot(Map conf, String id) {
+        if (mockedWorkerArtifactsRoot != null) {
+            return mockedWorkerArtifactsRoot;
+        }
+        return (workerArtifactsRoot(conf) + FILE_SEPARATOR + id);
+    }
+
+    public static String workerArtifactsRoot(Map conf, String id, Integer port) {
+        if (mockedWorkerArtifactsRoot != null) {
+            return mockedWorkerArtifactsRoot;
+        }
+        return (workerArtifactsRoot(conf, id) + FILE_SEPARATOR + port);
+    }
+
+    public static String workerArtifactsPidPath(Map conf, String id, Integer port) {
+        return (workerArtifactsRoot(conf, id, port) + FILE_SEPARATOR +  "worker.pid");
+    }
+
+    public static File getLogMetaDataFile(String fname) {
+        String[] subStrings = fname.split(FILE_SEPARATOR); // TODO: does this work well on windows?
```

Comment:
A jira filed for that in [STORM-1498](https://issues.apache.org/jira/browse/STORM-1498).